scripts/log2json

```diff
@@ -0,0 +1,195 @@
+#                 "Disk image(s) at /var/tmp/kstest-anaconda-conf.2022_03_04-03_50_44.h13p_9sy/disk-a.img,cache=unsafe",
+#                 "Results are in /var/tmp/kstest-anaconda-conf.2022_03_04-03_50_44.h13p_9sy"
+#             ],
+#             "result": "INFO: RESULT:anaconda-conf:ea89963f4984:SUCCESS:test done",
+#             "success": true,
+#             "name": "anaconda-conf",
+#             "end_time": 1646385568.0,
+#             "elapsed_time": 1724.0
+#         },
+# 
+#         ...
+# 
+#         {
+#             "scenario": "logs-daily-iso",
+#             "logfile": "INFO: install_log = /var/tmp/kstest-driverdisk-disk-kargs.2022_03_04-02_42_13._25nhiki/virt-install.log",
+#             "start_time": 1646379735.0,
+#             "errors": [
+#                 "ERROR: Install failed: Test timed out",
+#                 "ERROR: ERROR: Image creation failed: Test timed out"
+#             ],
+#             "result": "INFO: RESULT:driverdisk-disk-kargs:ea89963f4984:FAILED:Test timed out",
+#             "success": false,
+#             "name": "driverdisk-disk-kargs",
+#             "end_time": 1646383341.0,
+#             "elapsed_time": 3606.0
+#         }
+# ]
+
+import argparse
+import json
+import os
+import re
+from time import strptime, mktime
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Convert kstest logs to JSON")
+    parser.add_argument("--scenario", required=True, help="Name of test scenario")
+    parser.add_argument("--output", help="Output filename (default is stdout)")
+    parser.add_argument("LOGFILE", type=os.path.abspath, help="File to convert")
+
+    return parser.parse_args()
+
+# Skip lines that are not from tests, or that should be ignored
+def skip(line):
+    if "apply-ksappend" in line:
+        return True
+
+    # Ignore the virt-install cmdline
+    if "virt-install [" in line:
+        return True
+
+    return False
+
+def main(args):
+    # Read the file, a line at a time
+    # Start by ignoring lines
+    capture_summary = False
+    tests = []
+    entry = {}
+    stats = {"total": 0, "pass": 0, "fail": 0}
+
+    # Some assumptions about the logs are made:
+    # individual tests start with install_log INFO line and end with INFO RESULT line
+    with open(args.LOGFILE) as f:
+        for line in f.readlines():
+            line = line.strip()
+
+            if skip(line):
+                capture_summary = False
+                continue
+
+            # Parse the optional timestamp, the format is from python logging 'asctime'
+            m = re.match(r"(\d+-\d+-\d+\s\d+:\d+:\d+,\d+)", line)
+            timestamp = None
+            if m is not None:
+                try:
+                    timestamp = mktime(strptime(m.group(), "%Y-%m-%d %H:%M:%S,%f"))
+                except ValueError:
+                    pass
+
+            if m is not None:
+                line = line.removeprefix(m.group()).strip()
```

Comment:
repmoveprefix would be a problem running on runner having python 3.8 (https://github.com/rvykydal/kickstart-tests/runs/5689533247?check_suite_focus=true)